### DISCUSSION TITLE
Mask sensitive data in logs

### DISCUSSION THREAD

**Discussion by Piotr-Miller** [keycloak/keycloak#30404] [Q&A]
Hey,

I'm using Keycloak 24.0.4. Got questions about masking sensitive data in logs, for instance with *** or replacement text such as REDACTED_TOKEN.
Frankly speaking I did browse documentation and didn't find any information about it. 
My first thought was to manipulate option log-console-format while starting KC.
I will be grateful for any tip.

**Comment by orolhawion** [comment]
I could not find any configuration which would support this feature. I find it quite useful and so would like to ask kindly for implementation in the future.

**Comment by william-will-angi** [comment]
Would also like to see this, particularly around access logs. The login_hint query param is showing up plaintext in the logs

**Reply by shawkins** [reply]
@william-will-angi (and others) Logs, particularly above a debug level, emitted by Keycloak should be masked appropriately. If you have situation where you believe something sensitive is exposed incorrectly - please follow these instructions https://www.keycloak.org/security.html

As for masking in general there is support for it at the quarkus level https://github.com/quarkusio/quarkus/issues/33392
